src/modules/python/pbs/v1/_pmi_cray.py

```diff
@@ -538,4 +538,28 @@ def _pmi_ramp_up(self, hosts):
                         launch(func, cmd)
                         sleep_time = random.randint(1,10)
                         time.sleep(sleep_time)
-        return True
\ No newline at end of file
+        return True
+
+    def _pmi_power_status(self, hosts):
+        pbs.logmsg(pbs.LOG_DEBUG, "Cray: status of the nodes")
+        nidset = nodenids(hosts)
+        nids, _ = nidlist(None, nidset)
+        cmd = "node_status --nids " + nids
+        func = "pmi_power_status"
+        out = launch(func, cmd)
+        ready = []
+        nodeset = set()
+        if 'ready' in out:
+            ready = out['ready']
+        else:
+            return nodeset
+        craynid = "PBScraynid"
+        for vnames in hosts:
+            vnode = _svr_vnode(vnames)
+            try:
+                nid = int(vnode.resources_available[craynid])
+                if nid in ready:
+                    nodeset.add(vnames)
+            except Exception:
```

Comment:
The only thing that is going to raise an exception is possibly a KeyError.  You don't need the try/except.  You can just say if craynid in vnode.resources_available: above of where you check for it.

Reply:
Okay.